include/boost/beast/core/async_base.hpp

```diff
@@ -76,54 +76,73 @@ namespace beast {
 
     @code
 
-    // Asynchronously read into a buffer until the buffer is full, or an error occurs
-    template<class AsyncReadStream, class ReadHandler>
-    typename net::async_result<ReadHandler, void(error_code, std::size_t)>::return_type
-    async_read(AsyncReadStream& stream, net::mutable_buffer buffer, ReadHandler&& handler)
+    template<class DeducedHandler, class AsyncReadStream>
```

Comment:
A class template parameter is never labeled "deduced" I only use that for function parameters.